### PR TITLE
[IOTDB-4660] Implement IntoOperator & ITs

### PR DESCRIPTION
**Pull request by liuminghui233**
Related Documents:
- [User guide](https://apache-iotdb.feishu.cn/docx/doxcnqLcGFbsTBgUXG3eF8hFZxf)
- [Design doc](https://apache-iotdb.feishu.cn/docx/doxcnEPF2Sk9dyFzHdHPYIDehvc)

### CODE REVIEW

**Review by JackieTien97 — APPROVED** [review]
Remember to change the IT @Before to @BeforeClass

**Comment by liuminghui233** [comment]
> Remember to change the IT @before to @BeforeClass

fixed

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/DeviceViewIntoOperator.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            insertTabletStatementGenerators) {
+          lastReadIndex =
+              Math.max(lastReadIndex, generator.processTsBlock(inputTsBlock, readIndex));
+        }
+        readIndex = lastReadIndex;
+        insertMultiTabletsInternally(true);
+      }
+    }
+
+    if (child.hasNext()) {
```
Comment (yifuzhou) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/DeviceViewIntoOperator.java:102]: It is hard to understand here, in the end of the next() method we check child.hasNext()? I think check hasNext() should be called outside and when hasNext()=true, then do next().

**Review by yifuzhou — COMMENTED** [review]
PTAL~

File: server/src/main/java/org/apache/iotdb/db/client/DataNodeInternalClient.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    try {
+      sessionId =
+          SESSION_MANAGER.requestSessionId(
+              sessionInfo.getUserName(),
+              sessionInfo.getZoneId(),
+              IoTDBConstant.ClientVersion.V_0_13);
+
+      LOGGER.info("User: {}, opens internal Session-{}.", sessionInfo.getUserName(), sessionId);
+    } catch (Exception e) {
+      LOGGER.info("User {} opens internal Session failed.", sessionInfo.getUserName());
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/client/DataNodeInternalClient.java:83]: fixed

File: server/src/main/java/org/apache/iotdb/db/client/DataNodeInternalClient.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  public void close() {
+    SESSION_MANAGER.releaseSessionResource(sessionId, this::cleanupQueryExecution);
+    SESSION_MANAGER.closeSession(sessionId);
+  }
+
+  private void cleanupQueryExecution(Long queryId) {
+    IQueryExecution queryExecution = COORDINATOR.getQueryExecution(queryId);
+    if (queryExecution != null) {
+      try (SetThreadName threadName = new SetThreadName(queryExecution.getQueryId())) {
+        LOGGER.info("[CleanUpQuery]]");
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/client/DataNodeInternalClient.java:123]: fixed

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/DeviceViewIntoOperator.java (hunk 1 above)
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/DeviceViewIntoOperator.java:102]: fixed

**Pull request merged**